library/core/class.dispatcher.php

```diff
@@ -830,7 +841,9 @@ private function dispatchController($request, $routeArgs) {
      */
     private function createController($controllerName, $request, &$routeArgs) {
         /* @var Gdn_Controller $controller */
-        $controller = new $controllerName();
+        $controller = $this->container->get($controllerName);
+        // Allow classes to have a dependency on Gdn_Controller.
+        $this->container->setInstance('Gdn_Controller', $controller);
```

Comment:
I'd add a check to make sure that $controller is an instance of Gdn_Controller since that isn't a hard requirement.